allennlp/nn/util.py

```diff
@@ -1289,3 +1289,30 @@ def add_positional_features(tensor: torch.Tensor,
 def clone(module: torch.nn.Module, num_copies: int) -> torch.nn.ModuleList:
     "Produce N identical layers."
     return torch.nn.ModuleList([copy.deepcopy(module) for _ in range(num_copies)])
+
+
+def make2d(inputs: torch.Tensor) -> torch.Tensor:
+    """
+    Given a (possibly higher order) tensor of ids with shape
+    (batch_size, d1, ..., dn, sequence_length)
+    Return a view that's (batch_size * d1 * ... * dn, sequence_length).
+    """
+    if inputs.dim() > 2:
+        return inputs.view(-1, inputs.size(-1))
+    else:
+        return inputs
+
+def unmake2d(inputs: torch.Tensor, original_size: torch.Size) -> torch.Tensor:
+    """
+    Given a tensor of embeddings with shape
+    (batch_size * d1 * ... * dn, sequence_length, embedding_dim)
+    and the original shape
+    (batch_size, d1, ..., dn, sequence_length),
+    return the reshaped tensor of embeddings with shape
+    (batch_size, d1, ..., dn, sequence_length, embedding_dim)
+    """
+    if len(original_size) > 2:
+        view_args = list(original_size) + [inputs.size(-1)]
+        return inputs.view(*view_args)
+    else:
```

Comment:
Maybe this should error if the original size is < 2 and not equal to the dimension of the tensor?